central/clusterinit/service/service_impl.go

```diff
@@ -134,7 +136,17 @@ func (s *serviceImpl) GenerateCRS(ctx context.Context, request *v1.CRSGenRequest
 		return nil, status.Error(codes.Unimplemented, "support for generating Cluster Registration Secrets (CRS) is not enabled")
 	}
 
-	generated, err := s.backend.IssueCRS(ctx, request.GetName())
+	validUntil := fromProtoTimestamp(request.GetValidUntil())
+	validFor := request.GetValidFor().AsDuration()
+	if !validUntil.IsZero() && validFor != 0 {
```

Comment:
nit, absolutely not necessary, but something that reuses the existing `protocompat` function and also removes the need in lines 145-147.

```suggestion
	validUntil := protocompat.NilOrNow(request.GetValidUntil()).
		Add(request.GetValidFor().AsDuration())
	if request.GetValidUntil() != nil && request.GetValidFor() != nil {
```